corpus/data_augmentation.py

```diff
@@ -236,3 +243,167 @@ def run(self):
 
         with open(str(self.out_segment_file), "w") as segments_outfile:
             segments_outfile.writelines(segment_file_names)
+
+
+class AddImpulseResponseJob(Job):
+    """
+    Add impulse responses to each recording in the corpus with a given probability to be affected.
+    This is done by convolving the audio with an impulse response to simulate other acoustic conditions.
+    Supports .wav files
+
+    This job is dependent on the audiomentations library.
+    """
+
+    def __init__(
+        self, corpus_file, new_corpus_name, ir_path, augment_prob, time_rqmt=12
+    ):
+        """
+
+        :param Path corpus_file: Bliss corpus with wav files
+        :param str new_corpus_name: name of the new corpus
+        :param str ir_path: path to the directory containing impulse responses
+        :param float augment_prob: probability of each audio to be augmented by impulse response convolution
```

Comment:
Is it possible to make this randomness deterministic? (e.g. provide a seed to the job). Because otherwise it should be clearly marked that this job is not deterministic.

Reply:
Yes, both the randomness to choose an impulse response and the randomness to whether anything is done to a given audio come from the Python interpreter via the 'random' library. Setting the random.seed will make it deterministic. I'm not sure if I should put it in the job description string documentation, or add a new parameter to the call.